### ISSUE TITLE
Missing files in source package on PyPI

### ISSUE THREAD

**Issue by sagarkshah** [microsoft/msticpy#154]
In the [Source Dist on PyPI](https://files.pythonhosted.org/packages/f2/62/216ee6194fb79e197c17e60116d481a18019ed449277f1834d9d346b54a3/msticpy-1.0.0.tar.gz) the setup.py refers to requirements.txt and other files that are not present in the package causing the setup to fail.
```
     Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-download-4kdqlqg5/msticpy/setup.py", line 39, in <module>
        with open("requirements.txt", "r") as fh:
    FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'
```
Missing files probably need to be added to https://github.com/microsoft/msticpy/blob/master/MANIFEST.in

**Comment by ianhelle** [comment]
That seems logical but I don't understand how it's been installing for the past N months.
Should be fixed in #155. I will get a new version 1.0.1 to PyPI tomorrow.

**Comment by sagarkshah** [comment]
pip install works fine as is, locally. Our enterprise pipeline that allows use of external libraries in code uses setup.py to install the library. Also, looks like setup.py refers to "requirements-dev.txt" which is missing. 

I can provide more details over email/teams if required.

Thanks for taking a look at this.

**Comment by ianhelle** [comment]
Interesting. I've never used the --nobinary flag. It does fail when you specify that. I guess this for cautious types who want to compile any extensions in the package. We don't have any in msticpy but it should definitely be working even if this flag is specified. I'll double check the setup.py because I make reference to a couple of files like the README as well.
Thanks for the detail.

**Comment by ianhelle** [comment]
Resolved in #155. Published in version 1.0.1

**Comment by sagarkshah** [comment]
It seems to have bubbled up another error "pip._vendor.pep517.wrappers.BackendUnavailable". This is after running `pip  install --no-binary :all: msticpy==1.0.1`. I appreciate you taking a look, thanks!

**Comment by ianhelle** [comment]
What's the rationale for the --no-binary :all: flag? I think what you are doing may be difficult. 
I've repro'd issues with --no-binary flag on both Windows and Linux (although the precise messages are different depending on which package dependency fails). The --no-binary flag essentially says to pip - "don't download any pre-compiled wheels from PyPI, download the source and build the wheels locally".
The problem is not with msticpy itself but some of it's dependencies (like IPython, numpy and others) have C extensions. These need to be compiled locally, and for that you may also need statically-linked libraries that are not installed on the target system.

I'm pretty certain this is why it's failing.

- Are you installing into an environment where there are already Python packages installed or is this a blank slate (or new Python environment)?
- One thing you can try is taking the requirements.txt from this repo and running pip install -no-binary :all: -r requirements.txt. I think this will show you which package(s) is(are) failing.
- You can also try adding the pip --no-use-pep517, which forces legacy install

If you can try the second of these and the failures are common packages, you can usually apt-get install these prior to installing msticpy (e.g. sudo apt install python3-notebook jupyter jupyter-core). Most of the dependencies have an apt package on Ubuntu (or equivs on most Linux distros).

I'd be interested to hear more about the scenario. Happy to chat on Teams or whatever.

**Comment by ianhelle** [comment]
I've verified that if the requirements are installed `pip install --no-binary :all: msticpy==1.0.1` works without a problem. If the reason you are using the --no-binary flag is because you don't trust potential binaries in the PyPI wheels (not a bad thing), you may be able to use system installs of some of the python packages (like `sudo apt-get install python3-setuptools` - which is one of the packages that failed for me on Linux) prior to the pip install msticpy.

**Comment by zvezdan** [comment]
@ianhelle Yes, we have a full control of our build environment for Python packages and all the dependencies (direct, transitive, build system requires/setup_requires) are resolved and ready when rebuilding a package from source.
As long as the package declares its dependencies correctly and has all the files it uses in _setup.py_ (which seems to be fixed now in 1.0.1 release with my advice to refer to the files in _MANIFEST.in_) we can build it. I'll work with @sagarkshah offline to get this done.
Thank you!

**Comment by ianhelle** [comment]
Thanks @zvezdan and @sagarkshah. Please let me know if there's anything I can do to help. 
I'll leave this open for a while.
It might be good for others doing the same thing if you could post your solution (either the linux pkg install of the python packages or the libs used to build the Python package wheels).
